llvm-external-projects/torch-mlir-dialects/lib/Dialect/TMTensor/Transforms/Bufferize.cpp

```diff
@@ -0,0 +1,156 @@
+//===- Bufferize.cpp - Bufferization of tmtensor ops ------------------===//
```

Comment:
I am actually moving `torch-mlir-dialects` to `third_party` as suggested by Stella. Then it's more obvious this is code from third party. Would it be OK then if it contains code not used by IREE?

Reply:
It adds some maintanence burden cause the code needs to be kept in sync with LLVM. I'd rather only take those files that are needed for the conversion to `LinalgExt`

Reply:
If the build rules can be sliced so that we only actually depend on the dialect, then the extra files should not be a problem (ie. Allowing more of a straight CP). It would be better though if the upstream source were factored to be really thin, just providing the dialect.

If just the dialect, feel free to just include it in the main iree-opt and have a really simple smoke test along with the transform into IREE.

Reply:
> It would be better though if the upstream source were factored to be really thin, just providing the dialect.

Do you mean separate `lib/Dialect/TMTensor/IR (for dialects)` and `lib/Dialect/TMTensor/Transforms (for transforms)` into their own projects and include the dialects part for IREE?

Reply:
We probably dont need the entire `TMTensor` in IREE. We probably just need `TMTensor/IR`, and the pass here to convert from `TMTensor` to `LinalgExt`?

Reply:
got it. Removed everything else and only keep the TMTensor/IR.